### DISCUSSION TITLE
Android support?

### DISCUSSION THREAD

**Discussion by meganleewebb** [jglev/obsidian-apply-patterns-plugin#22] [Q&A] [answered]
Should this plugin work on Android? 

I'm trying to setup consistent rules to format files on laptop and mobile.

I have been testing it on Android.  Its a bit hard to enter the patterns on Android as the UI displays overly large. A lot of scrolling sideways needed. Some of the elements could shrink - like the 'disable' icon. Or stack them vertically instead when the screen space is small?

And then the patterns aren't all applied. 

eg

{
  "patterns": [
    {
      "name": "Whitespace",
      "done": false,
      "rules": [
        {
          "from": "[ \\t]+",
          "to": "#",
          "caseInsensitive": false,
          "global": true,
          "multiline": true,
          "sticky": false,
          "disabled": false
        }
      ],
      "collapsed": false
    }
  ],
  "filterString": "",
  "commandFilterString": "",
  "commands": [],
  "apiVersion": 3
}

A pattern will apply to a line, when when "apply to whole document", it doesn't work.

**Comment by jglev** [comment]
I'm sorry for my delay responding!

> I have been testing it on Android. Its a bit hard to enter the patterns on Android as the UI displays overly large. A lot of scrolling sideways needed. Some of the elements could shrink - like the 'disable' icon. Or stack them vertically instead when the screen space is small?

I agree with your assessment. Technologically, the plugin _should_ work on Android, but I haven't devoted any time to developing or using it on mobile yet. I'd certainly welcome contributions on styling, especially reactive (desktop- and mobile-compatible) styling! But in any case, this is on my mind -- improvements for usability on smaller screens is on my todo list. Thank you for surfacing it!

> And then the patterns aren't all applied.

I will look into this!

**Comment by meganleewebb** [comment]
Thanks for the response.
I use a sed script now. It's not triggered  from within obsidian, but gives me the same result on Android and laptop.

**Comment by jglev** [comment, marked as answer]
@meganleewebb, an update: the upcoming [`1.4.0` release](https://github.com/jglev/obsidian-apply-patterns-plugin/discussions/32) includes a newly-designed settings page that is (hopefully) more mobile-friendly, as well as several fixes for applying patterns specifically to whole documents.
